core/src/main/java/software/amazon/awssdk/async/AsyncResponseHandler.java

```diff
@@ -46,6 +49,8 @@
      * will be treated as a failure of the response and the {@link #exceptionOccurred(Throwable)} callback will be invoked.
      * </p>
      *
+     * <p>This callback may never be called if the response has no content or if an error occurs.</p>
```

Comment:
Oh sorry getting my interfaces mixed up. SdkHttpResponseHandler is the HTTP layer version of this. This interface only comes into play for streaming operations in which case it's adapted to the HTTP layer version. But it is possible to have no content and not have onStream called.